### ISSUE TITLE
Consider adding @pragma('vm:prefer-inline') hints across the board

### ISSUE THREAD

**Issue by shyndman** [Midi12/fnv#2]
Hello again,

I decided to try out hinting the compiler to inline the package's hash functions by using `@pragma('vm:prefer-inline')`, and it appears to have yielded a pretty substantial improvement in performance, at least in a Dart AOT build (VM build appears unaffected).

Would you be open to a PR?

**Comment by Midi12** [comment]
Hello,

Thanks for opening this issue.

Would the `@pragma('vm:prefer-inline')` hint only inline code in the callee? I guess yes but I did not find any official documentation about this particular hint online.

**Comment by shyndman** [comment]
> Would the @pragma('vm:prefer-inline') hint only inline code in the callee?

Yup. If you're curious, here's the [loop over the call sites](https://github.com/dart-lang/sdk/blob/eb17dad60440daeb5998751db8311d82769cfb64/runtime/vm/compiler/backend/inliner.cc#L952-L989), and the [pragma check](https://github.com/dart-lang/sdk/blob/eb17dad60440daeb5998751db8311d82769cfb64/runtime/vm/compiler/backend/inliner.cc#L2437-L2447). They meet in the middle somewhere. :)

> I guess yes but I did not find any official documentation about this particular hint online.

Me either. Ripe for misuse, I suspect.
